### ISSUE TITLE
Buildpack incompatible with building in /app

### ISSUE THREAD

**Issue by edmorley** [ducheneaut/miniconda-buildpack#1]
Hi

I'm on the team that maintains Heroku's build system and official buildpacks, and wanted to let you know about a future incompatibility with this buildpack.

The directory in which the Heroku build system performs builds is currently a path like`/tmp/build_<hash>`.

In the near future this path will be changing to `/app` so that the build-time and run-time app locations are the same path - in order to resolve a number of long standing bugs, and reduce the number of hacks buildpacks have to use to work around non-relocatable languages/toolchains.

Once this change comes into effect this buildpack will error like so:

```
remote: -----> Python/Miniconda app detected
remote: cp: '/app/requirements-conda.txt' and '/app/requirements-conda.txt' are the same file
remote: cp: '/app/requirements.txt' and '/app/requirements.txt' are the same file
remote: cp: '/app/stdlib' and '/app/stdlib' are the same file
remote: cp: '/app/test' and '/app/test' are the same file
remote:  !     Push rejected, failed to compile Python/Miniconda app.
```

This is because the buildpack tries to move the contents of `/app` to a temporary location, followed by moving BUILD_DIR (which is now /app) into `/app`.

To reproduce, the new behaviour can be enabled manually (ahead of it being the new default), using:

```
heroku labs:enable build-in-app-dir -a my_app_name
```

To fix, the file moves (at both the start and end of the build) need to be put behind a conditional that first checks if BUILD_DIR is already `/app`:
https://github.com/ducheneaut/miniconda-buildpack/blob/2b58a608c52b4d7fe80c5f51151f5864aff50998/bin/compile#L32-L36
https://github.com/ducheneaut/miniconda-buildpack/blob/2b58a608c52b4d7fe80c5f51151f5864aff50998/bin/compile#L159-L160

**Comment by ducheneaut** [comment]
Thanks Ed, appreciate the heads up. For planning, do you know when the new (incompatible) build system will be released?

**Comment by edmorley** [comment]
@ducheneaut Hi! Thank you for the fast reply :-) We're hoping to ship the change in the next week or two - it mainly depends on how many buildpacks are still incompatible at that point. Once the change ships, there will be a way to opt out.

**Comment by ducheneaut** [comment]
Got it. I'll roll out an update in the next few days to make sure we're ready.

**Comment by edmorley** [comment]
Fixed in ab925131e41869d8b4d8128cd6f68d98f90c18bc - thank you :-)
